### PR TITLE
Add uosmo, uion denom metadata

### PR DESCRIPTION
**Pull request by stackman27**
[Closes: #4224](https://github.com/osmosis-labs/osmosis/issues/4224)

## What is the purpose of the change
Add denom metadata for uosmo, uion

## Brief Changelog
- Add denom metadata for uosmo, uion

## Testing and Verifying
tested in upgrade handler 

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: app/upgrades/v15/upgrade_test.go [hunk 1]
```diff
@@ -66,3 +66,25 @@ func (suite *UpgradeTestSuite) TestMigrateNextPoolIdAndCreatePool() {
...
+	bankKeeper := suite.App.BankKeeper
+
+	// system under test.
+	v15.RegisterOsmoIonMetadata(ctx, *bankKeeper)
+
+	uosmoMetadata, found := suite.App.BankKeeper.GetDenomMetaData(ctx, "uosmo")
+	suite.Require().True(found)
+
+	uionMetadata, found := suite.App.BankKeeper.GetDenomMetaData(ctx, "uion")
+	suite.Require().True(found)
```
Comment (ValarDragon) [inline app/upgrades/v15/upgrade_test.go:86]: Can you test that these are false before the method is ran?

File: app/upgrades/v15/upgrades.go [hunk 2]
```diff
@@ -33,6 +37,9 @@ func CreateUpgradeHandler(
 		// See RunMigrations() for details.
 		fromVM[poolmanagertypes.ModuleName] = 0
 
+		// add metadata for uosmo and uion
```
Comment (ValarDragon) [inline app/upgrades/v15/upgrades.go:40]:
```suggestion
		// add metadata for uosmo and uion
		// These were missing before
```

File: app/upgrades/v15/upgrades.go [hunk 3]
```diff
@@ -53,3 +60,43 @@ func migrateNextPoolId(ctx sdk.Context, gammKeeper *gammkeeper.Keeper, poolmanag
...
+				Aliases:  nil,
+			},
+			{
+				Denom:    appParams.HumanCoinUnit,
+				Exponent: appParams.OsmoExponent,
+				Aliases:  nil,
+			},
+		},
+		Base:    appParams.BaseCoinUnit,
+		Display: appParams.HumanCoinUnit,
```
Comment (ValarDragon) [inline app/upgrades/v15/upgrades.go:80]: Confirmed that these AppParams are const's, so no need to worry about anything in build editing them

**Review by mattverse — COMMENTED** [review]
LGTM once @ValarDragon 's testing recommendation has been added

**Review by mattverse — COMMENTED** [review]
Can we add changelog entry for this

**Comment by stackman27** [comment]
addressed all comments!

File: app/upgrades/v15/upgrade_test.go [hunk 4]
```diff
@@ -66,3 +66,31 @@ func (suite *UpgradeTestSuite) TestMigrateNextPoolIdAndCreatePool() {
...
+func (suite *UpgradeTestSuite) TestRegisterOsmoIonMetadata() {
+	suite.SetupTest() // reset
+
+	expectedUosmodenom := "uosmo"
+	expectedUiondenom := "uion"
+
+	ctx := suite.Ctx
+	bankKeeper := suite.App.BankKeeper
+
+	uosmoMetadata, found := suite.App.BankKeeper.GetDenomMetaData(ctx, "uosmo")
```
Comment (mattverse) [inline app/upgrades/v15/upgrade_test.go:79]:
```suggestion
	// meta data should not be found pre-registration of meta data
	uosmoMetadata, found := suite.App.BankKeeper.GetDenomMetaData(ctx, "uosmo")
```

**Review by mattverse — APPROVED** [review]
LGTM

File: app/upgrades/v15/upgrade_test.go [hunk 5]
```diff
@@ -76,13 +76,19 @@ func (suite *UpgradeTestSuite) TestRegisterOsmoIonMetadata() {
 	ctx := suite.Ctx
 	bankKeeper := suite.App.BankKeeper
 
+	uosmoMetadata, found := suite.App.BankKeeper.GetDenomMetaData(ctx, "uosmo")
+	suite.Require().True(!found)
+
+	uionMetadata, found := suite.App.BankKeeper.GetDenomMetaData(ctx, "uion")
+	suite.Require().True(!found)
```
Comment (ValarDragon) [inline app/upgrades/v15/upgrade_test.go:83]:
```suggestion
	suite.Require().False(found)

	uionMetadata, found := suite.App.BankKeeper.GetDenomMetaData(ctx, "uion")
	suite.Require().False(found)
```
Annoying nit, sorry :sweat_smile:

**Review by ValarDragon — APPROVED** [review]
Awesome, thanks!

**Pull request merged**